File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/okhttp/ForwardingResponseBody.java
package com.didichuxing.doraemonkit.kit.network.okhttp;

import java.io.InputStream;

import okhttp3.MediaType;
import okhttp3.ResponseBody;
import okio.BufferedSource;
import okio.Okio;

public class ForwardingResponseBody extends ResponseBody {
    private final ResponseBody mBody;
    private final BufferedSource mInterceptedSource;

    public ForwardingResponseBody(ResponseBody body, InputStream interceptedStream) {
        mBody = body;
        mInterceptedSource = Okio.buffer(Okio.source(interceptedStream));
    }

    @Override
    public MediaType contentType() {
        return mBody.contentType();
    }

    @Override
    public long contentLength() {
        return mBody.contentLength();
    }

    @Override
    public BufferedSource source() {
        return mInterceptedSource;
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/okhttp/ForwardingResponseBody.kt
package com.didichuxing.doraemonkit.kit.network.okhttp

import okhttp3.MediaType
import okhttp3.ResponseBody
import okio.BufferedSource
import okio.Okio
import java.io.InputStream

class ForwardingResponseBody(private val mBody: ResponseBody, interceptedStream: InputStream?) : ResponseBody() {
    private val mInterceptedSource: BufferedSource = Okio.buffer(Okio.source(interceptedStream))
    override fun contentType(): MediaType? {
        return mBody.contentType()
    }

    override fun contentLength(): Long {
        return mBody.contentLength()
    }

    override fun source(): BufferedSource {
        return mInterceptedSource
    }

}